pytensor/tensor/rewriting/elemwise.py

```diff
@@ -380,6 +380,99 @@ def is_dimshuffle_useless(new_order, input):
     return is_useless
 
 
+@node_rewriter([Elemwise])
+def local_elemwise_lift_scalars(fgraph, node):
```

Comment:
This rewrite is not really lifting scalars, maybe call it "elemwise_to_scalar"?

Anyway, wouldn't it be easier to create a different Numba function when dispatching? Don't you still have the same problem with mixed rank0 and non rank0 inputs?

Reply:
Yes, elemwise_to_scalar is better :-)
I could have a separate numba function to solve the immediate scalar issue in elemwise...
I'll write a bit more about my motivation for this rewrite below...